compliance_checker/ioos.py

```diff
@@ -346,6 +348,353 @@ def check_units(self, ds):
+        if glb_gts_attr:
+            results.append(self._check_gts_ingest(glb_gts_attr, glb_msg))
+        else:
+            results.append(default_pass_result)
+
+        # check variables
+        var_msg = ("If provided, the attribute \"gts_ingest\" of variable \"{v}\" "
+                   "must be a string and its value must be one of "
+                   "\"true\" or \"false\"")
+        for v in ds.variables:
+            _attr = getattr(ds.variables[v], "gts_ingest", None)
+            if _attr:
+                print(_attr)
+                results.append((self._check_gts_ingest(_attr, var_msg.format(v=v))))
+            else:
+                results.append(default_pass_result)
+
+        return results
+
+    def check_instrument_variables(self, ds):
+        """
+        If present, the instrument_variable is one that contains additional
+        metadata about the instrument the data was collected with.
+
+        Args:
+            ds (netCDF4.Dataset): open Dataset
+
+        Returns:
+            list of Results
+        """
+
+        results = []
+        instr_vars = get_instrument_variables(ds)
+
+        # check for component, disciminant
+        for instr in instr_vars:
+            if instr in ds.variables:
+                compnt = getattr(ds.variables[instr], "component", None)
+                m = [f"component attribute of {instr} ({compnt}) must be a string"]
+                if compnt:
+                    results.append(
+                        Result(BaseCheck.MEDIUM, isinstance(compnt, str), "instrument_variable", m)
+                    )
+                else:
+                    results.append(Result(BaseCheck.MEDIUM, True, "instrument_variable", m))
+
+                disct = getattr(ds.variables[instr], "discriminant", None)
+                m = [f"discriminant attribute of {instr} ({disct}) must be a string"]
+                if disct:
+                    results.append(
+                        Result(BaseCheck.MEDIUM, isinstance(disct, str), "instrument_variable", m)
+                    )
+                else:
+                    results.append(Result(BaseCheck.MEDIUM, True, "instrument_variable", m))
+
+        return results
+
+    def check_qartod_variables_references(self, ds):
+        """
+        For any variables that are deemed QARTOD variables, check that they
+        contain the "references" attribute and that the value of the attribute
+        is a valid URL.
+
+        Args:
+            ds (netCDF4.Dataset): open Dataset
+
+        Returns:
+            list of Results
+        """
+
+        vldr = validators.Validator()
+        vldr.require_presence_of("scheme", "host")
+        vldr.allow_schemes("http", "https")
+
+        results = []
+        ctxt = "qartod_variable:references"
+        for v in ds.get_variables_by_attributes(standard_name=lambda x: x in self._qartod_std_names):
+            msg = f"\"references\" attribute for variable \"{v.name}\" must be a valid URL"
```

Comment:
We're still supporting Python < 3.6, so `str.format()` should be used instead of f-strings for compatibility reasons.